yolox-burn/src/model/head.rs

```diff
@@ -0,0 +1,236 @@
+
+        // Outputs for each feature map
+        let (outputs, shapes): (Vec<Tensor<B, 3>>, Vec<(usize, usize)>) = izip!(
+            features,
+            &self.stems,
+            &self.cls_convs,
+            &self.cls_preds,
+            &self.reg_convs,
+            &self.reg_preds,
+            &self.obj_preds,
+            &STRIDES
+        )
+        .map(
+            |(feat, stem, cls_conv, cls_pred, reg_conv, reg_pred, obj_pred, _stride)| {
+                let feat = stem.forward(feat);
+
+                let cls_feat = cls_conv.forward(feat.clone());
+                let cls_out = cls_pred.forward(cls_feat);
+
+                let reg_feat = reg_conv.forward(feat);
+                let reg_out = reg_pred.forward(reg_feat.clone());
+
+                let obj_out = obj_pred.forward(reg_feat);
+
+                // Output [B, 5 + num_classes, num_anchors]
+                let out = Tensor::cat(vec![reg_out, sigmoid(obj_out), sigmoid(cls_out)], 1);
+                let [_, _, h, w] = out.dims();
+                (out.flatten(2, 3), (h, w))
+            },
+        )
+        .unzip();
+
+        // 1. Concat all regression outputs
+        // 2. Permute shape to [B, num_anchors_total, 5 + num_classes]
+        // 3. Decode absolute bounding box values
+        self.decode(Tensor::cat(outputs, 2).swap_dims(2, 1), shapes.as_ref())
+    }
+
+    /// Decode bounding box absolute values from regression output offsets.
+    fn decode(&self, outputs: Tensor<B, 3>, shapes: &[(usize, usize)]) -> Tensor<B, 3> {
+        let device = outputs.device();
+        let [b, num_anchors, num_outputs] = outputs.dims();
+
+        let (grids, strides) = shapes
+            .iter()
+            .zip(STRIDES)
+            .map(|((h, w), stride)| {
+                // Grid (x, y) coordinates
+                let num_anchors = w * h;
+                let grid =
+                    create_2d_grid::<B>(*w, *h, &device).reshape(Shape::new([1, num_anchors, 2]));
+                let strides: Tensor<B, 3, Int> =
+                    Tensor::full(Shape::new([1, num_anchors, 1]), stride as i64, &device);
+
+                (grid, strides)
+            })
+            .unzip();
+
+        let grids = Tensor::cat(grids, 1).float();
+        let strides = Tensor::cat(strides, 1).float();
+
+        Tensor::cat(
+            vec![
+                // Add grid offset to center coordinates and scale to image dimensions
+                (outputs.clone().slice([0..b, 0..num_anchors, 0..2]) + grids) * strides.clone(),
+                // Decode `log` encoded boxes with `exp`and scale to image dimensions
+                outputs.clone().slice([0..b, 0..num_anchors, 2..4]).exp() * strides,
+                // Classification outputs
+                outputs.slice([0..b, 0..num_anchors, 4..num_outputs]),
+            ],
+            2,
+        )
+    }
+
+    // TODO
+    // Training outputs
+    // fn forward_train(&self, x: FpnFeatures<B>) {}
```

Comment:
Dead comment?

Reply:
Yeah I initially thought about implementing the forward for inference and train (which requires other intermediate outputs) so I left this TODO comment. But I think we should just start with the inference support and implement training in another PR.